Fix multiple issues with DynamicSolidMechanicsPhysics

bwspenc opened the pull request:
ref #31007

* Allow it to be used to set up weak plane strain models
* Correctly set use_displaced_mesh in the kernels (which required updating the gold file for one test, which ran slightly differently)
* Remove deprecated time integration parameters: alpha, beta, gamma, eta, zeta

To make these changes, I needed to determine whether a parameter existed in the params block. To address this, added this method: InputParameters::isParamDefined()

bwspenc committed:
Fix multiple issues with DynamicSolidMechanicsPhysics ref #31007

* Allow it to be used to set up weak plane strain models
* Correctly set use_displaced_mesh in the kernels (this required
  updating the gold files for several tests, all of which were set
  up with finite strain, and ran slightly differently)
* Remove deprecated time integration parameters: alpha, beta, gamma,
  eta, zeta

To make these changes, I needed to determine whether a parameter existed
in the params block. To address this, added this method:
InputParameters::isParamDefined()

GiudGiud commented on framework/include/utils/InputParameters.h:
```diff
@@ -967,11 +967,17 @@ class InputParameters : public libMesh::Parameters
    */
   bool isParamSetByUser(const std::string & name) const;
 
+  /**
```
this works. Let's put a note in both routines "if you dont know the type, use isParamDefined" and "if you know the type, use have_parameter" instead

bwspenc replied:
OK, will do. Sort of unrelated, but I'm not sure why `have_parameter` doesn't follow the camelcase style of methods in moose... Maybe it isn't used very widely -- I'll look.

bwspenc replied:
Looks like it is used widely, so I'm not going to touch it. 

I thought about naming this method `haveParameter()`, but I think `isParamDefined()` is more clear, because that only indicates that it has been defined, but not that it has been set.

dschwen replied:
have_parameter comes from libmesh

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3032522/), step Docs: sync website on 274eb22 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/31011/site)

This comment will be updated on new commits.

bwspenc committed:
Add tests for Dynamic SolidMechanics under various conditions ref #31007

bwspenc committed:
Add comments on isParamDefined to address review comment

bwspenc committed:
Add reference to paper in documentation of weak plane stress

bwspenc committed:
Update another set of tests affected by DynamicSolidMechanicsPhysics fix closes #31007

These tests were set up to use finite strain, but the
DynamicTensorMechanics action was not correctly setting the kernel up to
use finite strain. Regold for this bugfix.

I didn't catch these in the first pass because they required a larger
than default ad size. A few of them actually run with the default ad
size of 64, so I updated that in the tests file accordingly. There
was also a test that specified an ad size of 150, but it needed
a larger size.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3038837/), step Generate coverage on ec33076 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>9ec918</th>
      <th colspan=4>#31011 ec3307</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.53%</td>
      <td>85.53%</td>
      <td>-0.00%</td>
      <td>0.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>115343</td>
      <td>115343</td>
      <td>-</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>19516</td>
      <td>19520</td>
      <td>+4</td>
      <td>4</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/31011/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/31011/coverage/framework)


## Modules coverage

### Solid mechanics

<table>
  <thead>
    <tr>
      <th></th>
      <th>9ec918</th>
      <th colspan=4>#31011 ec3307</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.04%</td>
      <td>85.03%</td>
      <td>-0.01%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>28085</td>
      <td>28072</td>
      <td>-13</td>
      <td>13</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>4941</td>
      <td>4942</td>
      <td>+1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/31011/coverage_diff/solid_mechanics)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/31011/coverage/solid_mechanics)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/31011/coverage/xfem)

</details>

## Warnings

- `framework` new line coverage rate 0.00% is less than the suggested 90.0%



This comment will be updated on new commits.

bwspenc committed:
Add unit test for isParamDefined

GiudGiud reviewed (commented):
that's a  lots of regolding. Are they all from removing deprecation?

moosebuild commented:
Job [Public apps](https://civet.inl.gov/job/3039950/) on 17e1a19 : invalidated by @bwspenc

MASTODON PR that removes usage of deprecated syntax has been merged
